### DISCUSSION TITLE
Stream to a VLC renderer in a single invokation

### DISCUSSION THREAD

**Discussion by vivien** [streamlink/streamlink#4045] [Q&A]
I currently do `streamlink https://www.twitch.tv/twitch 720p` then click _Playback_ > _Renderer_ > _Foobar_ in VLC (where "Foobar" is the name of the chromecast on my network).

How can I invoke streamlink to stream on the chromecast in a single command?

**Comment by bastimeyer** [comment]
https://streamlink.github.io/latest/cli.html#player-options

https://wiki.videolan.org/VLC_command-line_help/
https://wiki.videolan.org/Documentation:Command_line/

**Reply by mkbloke** [reply]
In addition to the above, the relevant string for Streamlink's `-a` option which will contain the VLC options you're looking for, they are *probably* here: https://forum.videolan.org/viewtopic.php?f=16&t=143114#p477227

**Reply by vivien** [reply]
Thanks @mkbloke, but isn't the chromecast renderer being abstracted somehow? If the GUI provides simply Playback > Renderer > "Foobar", it feels weird to have to specify the chromecast IP and all on command line. Isn't there VLC command line options to specify a given renderer (whatever it is bound to)? (I mean something generic like an imaginary `--playback-renderer Foobar` option.)

**Reply by mkbloke** [reply]
It would appear that the name or device string that you see in the VLC GUI can't be used to specify the device via the command line, unfortunately.

**Reply by vivien** [reply]
Quite unfortunate indeed. Maybe VLC isn't a good fit to cast from streamlink directly. What do you guys would recommend to cast directly from streamlink to a given Chromecast device on your network?

**Reply by mkbloke** [reply]
I have nothing to recommend as it's not something I do.  There are probably other places where you can get better answers to that question.

**Comment by mkbloke** [comment]
One last thought on this, as it's way out of scope for Streamlink - if you can find the Chromecast device name via `avahi-browse -art` then you could perhaps do something like:
```
streamlink -O <url> <quality> | cvlc --sout "#chromecast" --sout-chromecast-ip=$(avahi-resolve-host-name <chromecast-name> | awk '{ print $2 }') --demux-filter=demux_chromecast
```
